src/utp_wrapper.rs

```diff
@@ -47,12 +48,20 @@ impl UtpWrapper {
                                     }
                                 },
                                 Err(TryRecvError::Disconnected) => break 'outer,
-                                Err(TryRecvError::Empty) => break,
+                                Err(TryRecvError::Empty) => {
+                                    attempts += 1;
+                                    if attempts < 5000 {
+                                        break;
+                                    } else {
+                                        break 'outer;
```

Comment:
this change doesn't look cool, as it'll limit us to always put something in the "write queue" if we don't drop the connection. doesn't look good.